### PR TITLE
docs: update documentation for validators package

### PR DESCRIPTION
**Pull request by SofiaMargariti**
## Purpose
Add documentation for @experiences/validators package

Screenshot 2024-05-17 at 16 27 12

### CODE REVIEW

**Comment by SofiaMargariti** [comment]
Hey @ryunsong-contentful, here is the updated documentation for the validators package, feel free to have a look :)

File: packages/validators/README.md [hunk 1]
```diff
@@ -1,7 +1,20 @@
...
+# @contentful/experiences-validators
 
 This package defines schemas and validator functions for experience data structures in the [Experiences SDK](https://www.contentful.com/developers/docs/experiences/set-up-experiences-sdk/). These validators ensure the integrity and correctness of the data used in experiences.
 
-## Documentation
+## Purpose
 
-Please refer to our [Documentation](https://www.contentful.com/developers/docs/experiences/data-structures/) to learn more about it.
+Defines and exports schema validators for experience and pattern entries as well for component definitions.
+Exports types related to experience fields infered from the schemas.
```
Comment (ryunsong-contentful) [inline packages/validators/README.md:8]: typo `infered` to `inferred`

File: packages/validators/README.md [hunk 2]
```diff
@@ -1,7 +1,20 @@
...
+
+
+## Concepts
+- **schemaVersion** Each experience or pattern entry contains the schema version it need to conform to under `fields` -> `componentTree` -> `schemaVersion`. We use this version to know which schema validations to run.
+- **schemas** Under the schemas folder we define the schema validations for each schema version we support. Can also contain other schema used
```
Comment (ryunsong-contentful) [inline packages/validators/README.md:13]: missing period at the end

File: packages/validators/README.md [hunk 3]
```diff
@@ -1,7 +1,20 @@
...
+in Studio Experiences, e.g. schema for the component definitions. All schemas are defined using the [Zod](https://zod.dev/) libary.
+- **validators** Functions that validate different entities, e.g. experiences using the schemas defined in this package.
+- **ExperienceEntryFields** This include only the reserved for experiences fields: `componentTree`, `dataSource`, `unboundValues`, `componentSettings` and `usedComponents`. These validators are not meant to validate any custom fields.
```
Comment (ryunsong-contentful) [inline packages/validators/README.md:16]: typo - `include` to `includes`

**Pull request merged**